cmake/scripts/ROOTConfig.cmake.in

```diff
@@ -76,7 +90,32 @@ foreach(_opt ${_root_options})
 endforeach()
 
 #----------------------------------------------------------------------------
-# Now set them to ROOT_LIBRARIES
+# Prepare utility interface targets
+add_library(ROOT::Libraries INTERFACE IMPORTED)
+
+add_library(ROOT::Flags IMPORTED INTERFACE)
```

Comment:
Ideally, yes, all targets should get either compile-options, compile-definitions, or compile-features set on them.

Reply:
> Ideally, yes, all targets should get either compile-options, compile-definitions, or compile-features set on them.

This is the direction we want to go, only adding the really required flags to each target, and actually not changing `CMAKE_CXX_FLAGS` directly at all ([see here](https://gist.github.com/mbinna/c61dbb39bca0e4fb7d1f73b0d66a4fd1#get-your-hands-off-cmake_cxx_flags) for more info).

Reply:
Compile features and C++ standard are not requirements. We should not force users to compile with the same standard as ROOT, in the same other libraries do not force us to compile ROOT with the same C++ standard as they were compiled. GeantV has had problems with ROOT because of this leakage of C++ standard flags in the past. @peremato can maybe comment on this.

Reply:
Yes, you are right. After my comment I remembered that ROOT has things like `RStringView.hxx`, so if users don't use the same standard, the backport will clash with `<string_view>` itself and wreck havoc. Same for other similar things, like for `make_unique`. The C++ standard flag probably needs to stay in there.